src/txn/txn.c

```diff
@@ -138,6 +138,50 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
     }
 }
 
+/*
+ * __wt_txn_in_global_list --
+ *     Check if a transaction is still in the global transaction list. If it is not, then it is
+ *     committed, prepared, or rolled back. It is possible that we race with commit, prepare or
+ *     rollback and a transaction is still in the global list before the start of the call is
+ *     reported not in the list.
+ */
+bool
+__wt_txn_in_global_list(WT_SESSION_IMPL *session, uint64_t txnid)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_TXN_GLOBAL *txn_global;
+    WT_TXN_SHARED *s;
+    uint64_t oldest_id;
+    uint32_t i, session_cnt;
+    bool in_global_list;
+
+    conn = S2C(session);
+    txn_global = &conn->txn_global;
+    in_global_list = true;
+
+    /* We're going to scan the table: wait for the lock. */
+    __wt_readlock(session, &txn_global->rwlock);
+    oldest_id = txn_global->oldest_id;
+
+    if (txnid <= oldest_id) {
```

Comment:
Just double checking that this should be "<=" and not "<"? If the `txn_id` is the current `oldest_id` then we should return true.

Reply:
Thanks for the comment. "<=" is correct, but it makes question the naming of the function again. Here we are checking whether the txn_id is behind the oldest id not against the ongoing transaction list.

Reply:
Sorry, you are right here we should use "<" instead as last_running may not be committed.